src/mediaelement.js

```diff
@@ -125,7 +125,7 @@ WaveSurfer.util.extend(WaveSurfer.MediaElement, {
     },
 
     getDuration: function () {
-        var duration = this.media.duration;
+        var duration = this.buffer.duration;
```

Comment:
The problem is, in MediaElement the buffer is optional. It's either
 * using Web Audio to decode a file (in which case the buffer is there)
 * simply plays the file without a waveform (no buffer).

So, maybe `var duration = (this.buffer || this.media).duration;`?